src/main/java/lukemccon/airdrop/packages/CreatePackageGui.java

```diff
@@ -0,0 +1,146 @@
+
+        int inventorySize = 36;
+        int packageCount = PackageManager.getNumberofPackages();
+
+        // Logic to determine how large to make the inventory
+        inventorySize = (int) (ROW_SIZE * (Math.ceil(((packageCount + SAVE_CANCEL_BACK_PADDING)/ROW_SIZE) + 1 ) + 1));
+
+        inv = Bukkit.createInventory(null, inventorySize, name);
+
+        initializeItems();
+
+        Bukkit.getPluginManager().registerEvents(this, Airdrop.PLUGIN_INSTANCE);
+    }
+
+    /**
+     * Setup control item blocks
+     */
+    public void initializeItems() {
+        int inventorySize = inv.getSize();
+
+        inv.setItem(inventorySize - 2, createGuiItem(Material.GREEN_WOOL, "Save"));
+        inv.setItem(inventorySize - 1 ,createGuiItem(Material.RED_WOOL, "Cancel"));
+    }
+    protected ItemStack createGuiItem(final Material material, final String name, final String... lore) {
+        final ItemStack item = new ItemStack(material, 1);
+        final ItemMeta meta = item.getItemMeta();
+
+        // Set the name of the item
+        meta.setDisplayName(name);
+
+        // Set the lore of the item
+        meta.setLore(Arrays.asList(lore));
+
+        item.setItemMeta(meta);
+
+        return item;
+    }
+
+    public void openInventory(final HumanEntity ent) {
+        ent.openInventory(inv);
+    }
+
+    @EventHandler
+    public void onInventoryClick(final InventoryClickEvent e) {
+        final Player p = (Player) e.getWhoClicked();
+
+        if (!e.getInventory().equals(inv)) return;
+
+        final ItemStack clickedItem = e.getCurrentItem();
+
+        if (clickedItem == null || clickedItem.getType().isAir()) return;
+
+        String itemStackName = clickedItem.getItemMeta().getDisplayName();
+
+        switch(itemStackName){
+            case "Save":
+                if (PermissionsHelper.isAdmin(p)) {
+                    this.save(e);
+                } else {
+                    ChatHandler.sendErrorMessage(p,"Must be admin to save edits to a package a package ");
+                    e.setCancelled(true);
+                }
+                break;
+
+            case "Cancel":
+                this.cancel(e);
+                break;
+
+            default:
+                if (!PermissionsHelper.isAdmin(p)) {
+                    e.setCancelled(true);
+                }
+        }
+    }
+
+    @EventHandler
+    public void onInventoryClick(final InventoryDragEvent e) {
+        Player p = (Player) e.getWhoClicked();
+        if (e.getInventory().equals(inv) && !PermissionsHelper.isAdmin(p)) {
+            e.setCancelled(true);
+        }
+    }
+
+    public String getName() { return this.name; }
+
+    public void save(final InventoryClickEvent e) {
+
+        Player p = (Player) e.getWhoClicked();
+
+        ItemStack[] newPackageItems = e.getInventory().getContents();
+        p.closeInventory();
+
+        pkg = new Package(this.name, this.price, new ArrayList<>(Arrays.asList(newPackageItems)));
```

Comment:
Is there any reason package specifically requires an ArrayList? I've seen a decent amount of places that you have to do ```ArrayList<>(Arrays.asList())``` because of it. Can we just switch the Package class to be of type List<?> ? Unless we're using any of the very specific ArrayList only methods it should be fine to just be a List.